services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/permissions/HankeKayttajaService.kt

```diff
@@ -1,21 +1,42 @@
 package fi.hel.haitaton.hanke.permissions
 
 import fi.hel.haitaton.hanke.HankeArgumentException
+import fi.hel.haitaton.hanke.application.ApplicationArgumentException
+import fi.hel.haitaton.hanke.application.ApplicationContactType
 import fi.hel.haitaton.hanke.application.ApplicationEntity
+import fi.hel.haitaton.hanke.application.ApplicationType
 import fi.hel.haitaton.hanke.configuration.Feature
 import fi.hel.haitaton.hanke.configuration.FeatureFlags
 import fi.hel.haitaton.hanke.domain.Hanke
 import fi.hel.haitaton.hanke.domain.Perustaja
+import fi.hel.haitaton.hanke.email.ApplicationInvitationData
 import fi.hel.haitaton.hanke.email.EmailSenderService
 import fi.hel.haitaton.hanke.email.HankeInvitationData
+import fi.hel.haitaton.hanke.getCurrentTimeUTC
 import fi.hel.haitaton.hanke.logging.HankeKayttajaLoggingService
+import fi.hel.haitaton.hanke.removeInviter
+import fi.hel.haitaton.hanke.typedContacts
+import fi.hel.haitaton.hanke.userContact
 import java.util.UUID
 import mu.KotlinLogging
 import org.springframework.stereotype.Service
 import org.springframework.transaction.annotation.Transactional
 
 private val logger = KotlinLogging.logger {}
 
+interface UserContact {
```

Comment:
This could be declared in a separate file. That file could also contain the two implementations and the stuff that's been added to Utils.kt.

This could be a sealed interface.

Reply:
Good suggestions, fixed.

Reply:
The extensions can be moved to the same file as well. I think Utils.kt should be reserved for highly generic things. Like things that are not directly related to any domain objects.

Reply:
Done. Also mover orderercount to applicationdata file.